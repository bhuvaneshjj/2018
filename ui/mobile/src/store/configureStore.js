import AsyncStorage from 'react-native'; // :TODO: use @react-native-community/async-storage
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import wsAPI from '../ui-common/store/wsAPI';
import { websocketConnected, websocketDisconnected } from '../ui-common/App/connectedActions';
import appReducer from '../App/appReducer';

export const loadState = async () => {
  try {
    const serializedState = await AsyncStorage.getItem('state');
    if (serializedState === null || serializedState === undefined) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while loading app state from the async storage', err);
    return undefined;
  }
};

const saveState = async state => {
  try {
    const serializedState = JSON.stringify(state);
    await AsyncStorage.setItem('state', serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while storing app state to the async storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onBroadcast: ({ broadcast }) => {
      switch (broadcast) {
        /* :TODO:
        case BROADCAST_UCASTNIK:
          store.dispatch(broadcastUcastnik(data));
          break;
        case BROADCAST_STOPKY:
          store.dispatch(broadcastStopky(data));
          break;
        */
        default:
          break;
      }
    },
    onConnect: async () => {
      await store.dispatch(websocketConnected());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam) => {
  const initialState = initialStateParam || {};

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(thunk.withExtraArgument(wsClient), wsAPI.withExtraArgument(wsClient))
    )
  );

  store.subscribe(async () => {
    const state = store.getState();
    await saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      },
      serviceURL: state.serviceURL
    });
  });

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
