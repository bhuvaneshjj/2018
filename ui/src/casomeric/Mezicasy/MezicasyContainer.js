import { connect } from 'react-redux';
import { startujiciNaTrase } from '../Startujici/StartujiciActions';
import { getDokoncenoWithCisloClass } from '../Startujici/startujiciReducer';
import StartCisloBox from '../StartCislo/StartCisloBox';
import StartCisloInputConnected from '../StartCislo/StartCisloInputContainer';
import Mezicasy from './Mezicasy';
import { addMezicas, removeMezicas } from './MezicasyActions';
import { getMezicasyWithCisloClass } from './mezicasyReducer';

export const computeMezicasy = state => {
  const startujici = getDokoncenoWithCisloClass(state.startujici, StartCisloBox);
  const mezicasy = getMezicasyWithCisloClass(state.casomeric.mezicasy, StartCisloInputConnected);

  const dohromady = startujici.concat(mezicasy);
  return dohromady.sort((a, b) => a.duration.valueOf() - b.duration.valueOf());
};

const mapStateToProps = state => ({
  mezicasy: computeMezicasy(state)
});

const mapDispatchToProps = dispatch => ({
  onRemoveMezicas: ({ id }) => {
    dispatch(removeMezicas(id));
  },
  onRemoveCislo: ({ id, duration }) => {
    dispatch(startujiciNaTrase(id));
    dispatch(addMezicas(duration));
  }
});

const mergeProps = (stateProps, dispatchProps) => ({
  mezicasy: stateProps.mezicasy.map(mezicas => {
    const onRemove = mezicas.cislo ? dispatchProps.onRemoveCislo : dispatchProps.onRemoveMezicas;
    return {
      ...mezicas,
      onRemove: () => onRemove(mezicas)
    };
  })
});

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(Mezicasy);
