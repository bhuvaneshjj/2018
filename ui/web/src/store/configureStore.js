import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { BROADCAST_STOPKY, BROADCAST_UCASTNIK } from 'ui-common/common';
import wsAPI from 'ui-common/store/wsAPI';
import appReducer from '../App/appReducer';
import { websocketConnected, websocketDisconnected } from '../App/AppActions';
import { broadcastStopky } from '../entities/stopky/stopkyActions';
import { broadcastUcastnik } from '../entities/ucastnici/ucastniciActions';
import { timesyncOperation, timesyncStart } from '../Timesync/TimesyncActions';

const loadState = () => {
  try {
    const serializedState = localStorage.getItem('state');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while loading app state from the local storage', err);
    return undefined;
  }
};

const saveState = state => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (err) {
    // :TODO: think about how to report the problem
    // eslint-disable-next-line no-console
    console.log('Problem while storing app state to the local storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onBroadcast: ({ broadcast, data }) => {
      switch (broadcast) {
        case BROADCAST_UCASTNIK:
          store.dispatch(broadcastUcastnik(data));
          break;
        case BROADCAST_STOPKY:
          store.dispatch(broadcastStopky(data));
          break;
        default:
          break;
      }
    },
    onConnect: async () => {
      await store.dispatch(websocketConnected());
      store.dispatch(timesyncOperation());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam = loadState()) => {
  const initialState = initialStateParam || {};

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(thunk.withExtraArgument(wsClient), wsAPI.withExtraArgument(wsClient))
    )
  );

  store.subscribe(() => {
    const state = store.getState();
    saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      }
      // :TODO: casomeric.mezicasy
    });
  });

  // The time synchronization gets started by default.
  store.dispatch(timesyncStart());

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
