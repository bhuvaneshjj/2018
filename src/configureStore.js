import { createStore } from 'redux';
import appReducer from './App/appReducer';
import { setHighestMezicasId } from './Mezicasy/MezicasyActions';

const preloadedState = {
  startujici: [
    { id: 0, cislo: 7, dokonceno: null },
    { id: 1, cislo: 4, dokonceno: null },
    { id: 10, cislo: 16, dokonceno: null },
    { id: 2, cislo: 25, dokonceno: false },
    { id: 3, cislo: 9, dokonceno: null },
    { id: 12, cislo: 15, dokonceno: null },
    { id: 9, cislo: 1, dokonceno: false },
    { id: 29, cislo: 8, dokonceno: null },
    { id: 5, cislo: 59, dokonceno: null },
    { id: 11, cislo: 43, dokonceno: null },
    { id: 18, cislo: 42, dokonceno: false },
    { id: 10, cislo: 33, dokonceno: null },
    { id: 7, cislo: 21, dokonceno: false },
    { id: 13, cislo: 24, dokonceno: null },
    { id: 34, cislo: 25, dokonceno: null },
    { id: 14, cislo: 22, dokonceno: null },
    { id: 58, cislo: 30, dokonceno: null },
    { id: 15, cislo: 23, dokonceno: null },
    { id: 59, cislo: 26, dokonceno: null }
  ]
};

const configureStore = () => {
  let highestId = 0;
  if (preloadedState.mezicasy) {
    preloadedState.mezicasy.forEach(mezicas => {
      if (mezicas.id > highestId) {
        highestId = mezicas.id;
      }
    });
  }

  setHighestMezicasId(highestId);

  return createStore(
    appReducer,
    preloadedState,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  );
};

export default configureStore;
