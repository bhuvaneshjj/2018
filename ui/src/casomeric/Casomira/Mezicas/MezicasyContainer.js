import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import moment from 'moment';
import withResponsive from '../../../shared/withResponsive/withResponsive';
import { getUcastiProRok } from '../../../entities/ucastnici/ucastniciReducer';
import { getMezicasy, getStopkyByTyp } from '../../Stopky/StopkyProTyp/stopkyProTypReducer';
import { saveStopky, stopkyRemoveMezicas } from '../../Stopky/StopkyProTyp/StopkyProTypActions';
import Mezicasy from './Mezicasy';

const mapStateToProps = (state, ownProps) => {
  const { typ } = ownProps;
  const stopky = getStopkyByTyp({ state, typ });
  const ucasti = getUcastiProRok({ ...state.entities });

  const mezicasy = getMezicasy({ stopky, ucasti, kategorie: state.entities.kategorie });
  const populated = mezicasy.map(mezicas => {
    const { cas, ...rest } = mezicas;
    return { ...rest, cas: moment.duration(cas) };
  });

  return { mezicasy: populated, typ };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  const { typ } = ownProps;

  return {
    onStopkyRemoveMezicas: ({ cas }) =>
      dispatch(saveStopky({ action: stopkyRemoveMezicas({ cas }), typ }))
  };
};

const mergeProps = (stateProps, dispatchProps) => {
  const { mezicasy, ...restOfStateProps } = stateProps;
  const { onStopkyRemoveMezicas, onUcastnikRemoveCas, ...restOfDispatchProps } = dispatchProps;

  const populated = mezicasy.map(mezicas => {
    const { id, cas } = mezicas;
    return {
      ...mezicas,
      onEdit: () => {},
      onRemove: id ? () => onUcastnikRemoveCas({ id }) : () => onStopkyRemoveMezicas({ cas })
    };
  });

  return {
    ...restOfStateProps,
    mezicasy: populated,
    ...restOfDispatchProps
  };
};

const MezicasyResponsive = withResponsive(Mezicasy, { disableWidth: true });

const MezicasyContainer = connect(mapStateToProps, mapDispatchToProps, mergeProps)(
  MezicasyResponsive
);

MezicasyContainer.propTypes = {
  typ: PropTypes.string.isRequired
};

export default MezicasyContainer;
