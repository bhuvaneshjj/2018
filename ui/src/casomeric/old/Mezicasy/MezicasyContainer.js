import { connect } from 'react-redux';
import { startujiciNaTrase } from '../Startujici/StartujiciActions';
import { getDokoncenoWithCisloClass } from '../Startujici/startujiciReducer';
import StartCisloBox from '../StartCislo/StartCisloBox';
import StartCisloInputContainer from '../StartCislo/StartCisloInputContainer';
import Mezicasy from './Mezicasy';
import { addMezicas, removeMezicas } from './MezicasyActions';
import { getMezicasyWithCisloClass } from './mezicasyReducer';

export const computeMezicasy = ({ casomeric: { maraton } }) => {
  const startujici = getDokoncenoWithCisloClass(maraton.startujici, StartCisloBox);
  const mezicasy = getMezicasyWithCisloClass(maraton.mezicasy, StartCisloInputContainer);

  const dohromady = startujici.concat(mezicasy);
  return dohromady.sort((a, b) => a.duration.valueOf() - b.duration.valueOf());
};

const mapStateToProps = state => ({
  mezicasy: computeMezicasy(state)
});

const mapDispatchToProps = dispatch => ({
  onRemoveMezicas: ({ id }) => {
    dispatch(removeMezicas(id));
  },
  onRemoveCislo: ({ id, duration }) => {
    dispatch(startujiciNaTrase(id));
    dispatch(addMezicas(duration));
  }
});

const mergeProps = (stateProps, dispatchProps) => ({
  mezicasy: stateProps.mezicasy.map(mezicas => {
    const onRemove = mezicas.cislo ? dispatchProps.onRemoveCislo : dispatchProps.onRemoveMezicas;
    return {
      ...mezicas,
      onRemove: () => onRemove(mezicas)
    };
  })
});

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(Mezicasy);
