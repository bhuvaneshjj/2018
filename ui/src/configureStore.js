import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import appReducer from './App/appReducer';
import { websocketConnected, websocketDisconnected } from './App/AppActions';
import { setHighestMezicasId } from './casomeric/Mezicasy/MezicasyActions';

const demoStartujiciState = {
  startujici: [
    { id: 0, cislo: 7, dokonceno: null },
    { id: 1, cislo: 4, dokonceno: null },
    { id: 17, cislo: 16, dokonceno: null },
    { id: 2, cislo: 25, dokonceno: false },
    { id: 3, cislo: 9, dokonceno: null },
    { id: 12, cislo: 15, dokonceno: null },
    { id: 9, cislo: 1, dokonceno: false },
    { id: 29, cislo: 8, dokonceno: null },
    { id: 5, cislo: 59, dokonceno: null },
    { id: 11, cislo: 43, dokonceno: null },
    { id: 18, cislo: 42, dokonceno: false },
    { id: 10, cislo: 33, dokonceno: null },
    { id: 7, cislo: 21, dokonceno: false },
    { id: 13, cislo: 24, dokonceno: null },
    { id: 34, cislo: 27, dokonceno: null },
    { id: 14, cislo: 22, dokonceno: null },
    { id: 58, cislo: 30, dokonceno: null },
    { id: 15, cislo: 23, dokonceno: null },
    { id: 59, cislo: 26, dokonceno: null }
  ]
};

const loadState = () => {
  try {
    const serializedState = localStorage.getItem('state');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    console.log('Problem while loading app state from the local storage', err);
    return undefined;
  }
};

const saveState = state => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (err) {
    console.log('Problem while storing app state to the local storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onConnect: () => {
      store.dispatch(websocketConnected());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam = loadState()) => {
  const initialState = initialStateParam || {};
  initialState.startujici = demoStartujiciState.startujici;

  if (initialState.casomeric && initialState.casomeric.mezicasy) {
    let highestId = 0;

    initialState.casomeric.mezicasy.forEach(mezicas => {
      if (mezicas.id > highestId) {
        highestId = mezicas.id;
      }
    });

    setHighestMezicasId(highestId);
  }

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(applyMiddleware(thunk.withExtraArgument(wsClient)))
  );

  store.subscribe(() => {
    const state = store.getState();
    saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      },
      casomeric: { stopky: state.casomeric.stopky, mezicasy: state.casomeric.mezicasy }
    });
  });

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
