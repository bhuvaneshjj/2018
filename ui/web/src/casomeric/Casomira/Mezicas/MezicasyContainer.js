import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import moment from 'moment';
import withResponsive from '../../../shared/withResponsive/withResponsive';
import { getUcastiProRok } from '../../../entities/ucastnici/ucastniciReducer';
import { getMezicasy, getStopkyByTyp } from '../../Stopky/StopkyProTyp/stopkyProTypReducer';
import {
  stopkyInsertMezicas,
  stopkyRemoveMezicas,
} from '../../Stopky/StopkyProTyp/StopkyProTypActions';
import {
  createDropAction,
  saveVykon,
  startCisloNaTrase,
} from '../StartovniCisla/StartovniCislaActions';
import { canDrop } from './MezicasyActions';
import Mezicasy from './Mezicasy';

const mapStateToProps = (state, ownProps) => {
  const { typ } = ownProps;
  const stopky = getStopkyByTyp({ state, typ });
  const ucasti = getUcastiProRok({ ...state.entities });

  const mezicasy = getMezicasy({ stopky, ucasti, kategorie: state.entities.kategorie });
  const populated = mezicasy.map((mezicas) => {
    const { cas, ...rest } = mezicas;
    return { ...rest, cas: moment.duration(cas) };
  });

  return { mezicasy: populated, typ };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  const { typ } = ownProps;
  const onStopkyRemoveMezicas = ({ cas }) => dispatch(stopkyRemoveMezicas({ cas, typ }));

  return {
    onDrop: (dropResult) => {
      dispatch(createDropAction(dropResult));
      onStopkyRemoveMezicas({ cas: dropResult.destination.cas });
      return undefined;
    },
    onStopkyRemoveMezicas,
    onUcastnikRemoveCas: ({ id, cas }) => {
      dispatch(stopkyInsertMezicas({ cas, typ }));
      dispatch(saveVykon({ action: startCisloNaTrase({ id }), id, typ }));
    },
  };
};

const mergeProps = (stateProps, dispatchProps) => {
  const { mezicasy, ...restOfStateProps } = stateProps;
  const {
    onDrop,
    onStopkyRemoveMezicas,
    onUcastnikRemoveCas,
    ...restOfDispatchProps
  } = dispatchProps;

  const populated = mezicasy.map((mezicas) => {
    const { id, cas } = mezicas;
    return {
      ...mezicas,
      canDrop,
      onDrop,
      onEdit: () => {},
      onRemove: id ? () => onUcastnikRemoveCas({ id, cas }) : () => onStopkyRemoveMezicas({ cas }),
    };
  });

  return {
    ...restOfStateProps,
    mezicasy: populated,
    ...restOfDispatchProps,
  };
};

const MezicasyResponsive = withResponsive(Mezicasy, { disableWidth: true });

const MezicasyContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
  mergeProps
)(MezicasyResponsive);

MezicasyContainer.propTypes = {
  typ: PropTypes.string.isRequired,
};

export default MezicasyContainer;
