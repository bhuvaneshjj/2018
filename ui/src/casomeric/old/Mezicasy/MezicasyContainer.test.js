import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import MezicasyContainer from './MezicasyContainer';

const mockStore = configureStore();

let store;
let wrapper;
beforeEach(() => {
  const state = {
    casomeric: {
      maraton: {
        mezicasy: [{ id: 0, duration: 'PT4.950S' }, { id: 1, duration: 'PT12.101S' }],
        startujici: [
          { id: 0, cislo: 8, dokonceno: true, duration: 'PT2H32M14.2S' },
          { id: 1, cislo: 7, dokonceno: null },
          { id: 2, cislo: 5, dokonceno: null },
          { id: 3, cislo: 25, dokonceno: false }
        ]
      }
    }
  };
  store = mockStore(state);
  store.dispatch = jest.fn();
  wrapper = shallow(<MezicasyContainer store={store} />);
});

it('maps state and dispatch to props', () => {
  expect(wrapper.props().mezicasy).toBeTruthy();
  expect(wrapper.props().mezicasy).toMatchSnapshot();
});

it('maps [0].onRemove to dispatch onRemoveMezicas action', () => {
  wrapper.props().mezicasy[0].onRemove();

  expect(store.dispatch).toHaveBeenCalledWith({ id: 0, type: 'REMOVE_MEZICAS' });
});

it('maps [2].onRemove to dispatch onRemoveMezicas action', () => {
  wrapper.props().mezicasy[2].onRemove();

  expect(store.dispatch).toHaveBeenCalledTimes(2);
  expect(store.dispatch).toHaveBeenLastCalledWith({
    id: 0,
    type: 'ADD_MEZICAS',
    duration: 'PT2H32M14.2S'
  });
});
