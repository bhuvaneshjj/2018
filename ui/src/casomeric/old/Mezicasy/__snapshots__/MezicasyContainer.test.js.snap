// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maps state and dispatch to props 1`] = `
Array [
  Object {
    "cisloClass": [Function],
    "duration": "PT4.95S",
    "id": 0,
    "onRemove": [Function],
  },
  Object {
    "cisloClass": [Function],
    "duration": "PT12.101S",
    "id": 1,
    "onRemove": [Function],
  },
  Object {
    "cislo": 8,
    "cisloClass": [Function],
    "duration": "PT2H32M14.2S",
    "id": 0,
    "onRemove": [Function],
  },
]
`;
